content/post/why-16k-page-size/index.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Why 16k page size matters
+description: 16k vs 4k page sizes & Asahi Linux.
+date: 2024-06-09 18:27:00 +0800
+---
+
+Since August of 2023, I've been a daily user of a 14-inch 2021 Apple MacBook Pro (M1 Pro, 16GB RAM) laptop.
+Thanks to [@tpwrules](https://github.com/tpwrules)'s [NixOS Asahi Linux Flake](https://github.com/tpwrules/nixos-apple-silicon/tree/main),
+I was able to set up the laptop to run NixOS. The majority of software have worked perfectly on it and I
+consider it to be the best Linux laptop for its performance. However, one major issue isn't a hardware issue
+but a software one. When you see memory sizes on a system, you commonly think about things like the timing,
+what kind of RAM chips they use, the number of RAM chips they use, and so on. Though, this isn't that.
+This particular issue is one designed into the Asahi Linux kernel and even macOS's Darwin kernel.
+This would be the page size, the number which dictates the minimum size of a page of memory is. Every
+modern computer nowadays, since the 90's, use something called page memory. This is a technique to
+efficiently manage memory but it is also used as a security mechanism. Programs can be given permissions
+to only certain regions of memory via page memory. However, this exact issue is not an issue with the security
+but the size.
+
+The vast majority of users use x86 which utilizes 4k page memory. This size was great when page memory
+was first introduced along with Windows 95. However, our memory sizes nowadays are vastly larger.
```

Comment:
`was first introduced along with Windows 95` introduced in the Win95 era or explicitly *with* Win95? I assume the former

Reply:
Win95 afaik is the first major consumer version of Windows to utilize protected mode and user mode which involves paging. 32-bit x86 CPU's specifically.